lib_sw_dac/src/standard_fidelity/sigma_delta_1_5.S

```diff
@@ -84,7 +105,33 @@ FUNCTION_NAME:
 #endif
     
     { bt  r9, .Linner_loop       ; ldaw dp, dp[SDAC_BUF_STEP] }
-    { bu  .Louter_loop           ; in r11, res[r0] }
+    waiteu
+
+    // Select handler for timeout. This does not need to speed optimised
+    .align 8
+.sd_timeout:
+    // Get the out dummy word, output it and then invert it to avoid DC output content
+    ldw r11, sp[SDAC_STACK_TIMEOUT_WORD];
+#if defined(SW_DAC_SD_TEST_MODE)
+    out res[r7], r11
+    out res[r8], r11
+#else
+    outpw res[r7], r11, 16;
+    outpw res[r8], r11, 16;
+#endif
+    not r11, r11;
+    stw r11, sp[SDAC_STACK_TIMEOUT_WORD];
+
+    // Move the timeout timer on by at least one PWM period. We may come back here if no IN from filter
+    { gettime r11               ; ldw r6, sp[SDAC_STACK_TIMEOUT_PERIOD]}
+    { add r11, r11, r6          ; ldw r6, sp[SDAC_STACK_TIMEOUT_RESID]}
+    { setd res[r6], r11         ; ldc r11, 0x1;} // Move time trigger forwards and load timeout flag
+    
+    // Set timeout occurred flag in struct
+    ldw r6, sp[SDAC_STACK_SD_STRUCT_BASE]   ;
+    {stw r11, r6[SDAC_TIMEOUT_OCCURED]      ; waiteu}
```

Comment:
I wonder why do we need the timeout_occured in the state struct if we don't handle it on the library level?
If we only want to use it for the test, I'd suggest removing it from the state and getting information to the test in a different way, if possible.

Reply:
This is quite a handy feature. We use something similar in I2S slave - when integrating you might find a case where you missed timing (system backpressure for example) and it's great to check a flag in that case.